server.go

```diff
@@ -219,9 +217,13 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 
 	ret.EventWorker = NewEventWorker(ret.Statsd)
 
-	ret.UDPAddr, err = net.ResolveUDPAddr("udp", conf.UdpAddress)
-	if err != nil {
-		return
+	for _, addrStr := range conf.StatsdListenAddresses {
+		var addr *ListeningAddr
+		addr, err = ResolveAddr(addrStr)
+		if err != nil {
+			return
```

Comment:
Let's just use an explicit return here; it also means we can avoid the extra `var` declaration a few lines up

Reply:
Can't, as far as I can tell; shadowing `err` is fine, but the linter complains:

``` go
		addr, err := ResolveAddr(addrStr)
		if err != nil {
			return ret, err  // <- return copies lock value: veneur.Server contains sync.Mutex
		}
		ret.StatsdListenAddrs = append(ret.StatsdListenAddrs, addr)
```

So I guess this is how we get to do it /:

Reply:
Alternative:
``` go
		addr, e := ResolveAddr(addrStr)
		if e != nil {
			err = e
			return
		}
		ret.StatsdListenAddrs = append(ret.StatsdListenAddrs, addr)
```

but that seems... unidiomatic? (I had to do a few double-takes checking which `e` and which `err` is right)

Reply:
That... looks like a linter bug. I mean, the return copies the mutex whether it's a named return or an explicit return, so I don't know why it would complain on the first but not the second.


I'd rather just use an explicit return here (since eventually we want to remove named returns from this function, but that's a yak we can shave later).

Reply:
SG. I'll go for the first one, then.